freshmaker/image_verifier.py

```diff
@@ -22,120 +22,119 @@
+            raise ValueError("'brew.build' field not set in image")
+
         if not image["content_sets"]:
             raise ValueError(
-                "Found image \"%s\" in this repository, but it cannot be rebuilt, because "
-                "the \"content_sets\" are not set for this image." % image.nvr)
+                'Found image "%s" in this repository, but it cannot be rebuilt, because '
+                'the "content_sets" are not set for this image.'
+                % image["brew"]["build"]
+            )
 
     def _get_repository_from_name(self, repo_name):
         """
         Returns the ContainerRepository object based on the Repository name.
         """
-        query = {
-            "objectType": "containerRepository",
-            "query": {
-                "$and": [
-                    {
-                        "field": "repository",
-                        "op": "=",
-                        "rvalue": repo_name
-                    },
-
-                ]
-            },
-            "projection": [
-                {"field": "*", "include": True, "recursive": True}
-            ]
-        }
+        repos = self.pgql.find_repositories_by_repository_name(repo_name)
 
-        repos = self.lb.find_container_repositories(query, auto_rebuild=False)
         if not repos:
-            raise ValueError("Cannot get repository %s from Lightblue." % repo_name)
+            raise ValueError("Cannot get repository %s from Pyxis." % repo_name)
+
         if len(repos) != 1:
-            raise ValueError("Multiple records found in Lightblue for repository %s." % repo_name)
+            raise ValueError(
+                "Multiple records found in Pyxis for image repository %s." % repo_name
+            )
 
         return repos[0]
 
-    def _get_repository_from_image(self, nvr):
+    def _get_repository_from_image(self, image):
         """
-        Returns the ContainerRepository object based on the image NVR.
+        Returns the ContainerRepository object based on the image defined by the image NVR.
         """
-        query = {
-            "objectType": "containerRepository",
-            "query": {
-                "$and": [
-                    {
-                        "field": "images.*.brew.build",
-                        "op": "=",
-                        "rvalue": nvr
-                    },
-
-                ]
-            },
-            "projection": [
-                {"field": "*", "include": True, "recursive": True}
-            ]
-        }
+        nvr = image["brew"]["build"]
 
-        repos = self.lb.find_container_repositories(query)
-        if not repos:
-            raise ValueError("Cannot get repository for image %s from Lightblue." % nvr)
-        if len(repos) != 1:
+        if "repositories" not in image or not image["repositories"]:
+            raise ValueError("Cannot get repository for image %s from Pyxis." % nvr)
+
+        if len(image["repositories"]) != 1:
```

Comment:
unlike the querying LightBlue repo way which should always returns a single repository (at least for redhat repositories), when you check published image's repositories it should always have at least 2 repositories, one is the internal OSBS build repo, so can't perform such check.